ansible/configs/studentvm/software.yml

```diff
@@ -0,0 +1,54 @@
+---
+- name: Step 004 - Software
+  hosts: localhost
+  gather_facts: False
+  become: false
+  tasks:
+  - debug:
+      msg: "Software tasks started"
+
+- name: Set up StudentVM
+  hosts: bastions
+  gather_facts: false
+  become: true
+  tasks:
+  - name: Run Student VM roles
+    when:
+    - studentvm_roles | d("") | length > 0
+    block:
+    - name: Run Student VM Role
+      include_role:
+        name: "{{ studentvm_role }}"
+      loop: "{{ studentvm_roles }}"
+      loop_control:
+        loop_var: studentvm_role
+
+  - name: Install FTL
+    when: >-
+      ftl_install | d(False) | bool or
```

Comment:
This is kind of in the middle of its evolution. You have `ftl_install` set to false (as well as default here), but you have `ftl_injector_tag` defined, so it will run. The idea was to drop the `ftl_install` completely and let the `ftl_injector` var decide if anything happens or not, but that got sidetracked and then there was more attention on fixing FTL wholistically.

So, a long way of saying, consider dropping one of these or cleaning up the default vars so it is not as confusing where you say `ftl_install=false`, but by defining the injector tag this evaluates true.

Reply:
OK. Removing install_ftl